tests/Aspire.Hosting.Tests/Azure/AzureBicepProvisionerTests.cs

```diff
@@ -84,6 +84,38 @@ public async Task ResourceWithTheSameBicepTemplateAndParametersHaveTheSameCheckS
         Assert.Equal(checkSum0, checkSum1);
     }
 
+    [Theory]
+    [InlineData("1alpha")]
+    [InlineData("-alpha")]
+    [InlineData("")]
+    [InlineData(" alpha")]
+    [InlineData("alpha 123")]
+    public void WithParameterDoesNotAllowParameterNamesWhichAreInvalidBicepIdentifiers(string bicepParameterName)
+    {
+        var ex = Assert.Throws<ArgumentException>(() =>
+        {
+            var builder = DistributedApplication.CreateBuilder();
+            builder.AddAzureConstruct("construct", _ => { })
+                   .WithParameter(bicepParameterName);
+        });
+
+        Assert.Equal("Bicep parameter names must only contain alpha, numeric, and _ characters with a leading alpha character. (Parameter 'bicepParameterName')", ex.Message);
+    }
+
+    [Theory]
+    [InlineData("alpha")]
+    [InlineData("a1pha")]
+    [InlineData("_alpha")]
+    [InlineData("__alpha")]
+    [InlineData("alpha1_")]
+    [InlineData("Alpha1_A")]
+    public void WithParameterAllowsParameterNamesWhichAreValidBicepIdentifiers(string bicepParameterName)
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        builder.AddAzureConstruct("construct", _ => { })
```

Comment:
I wish we could shift-left the use-case when a parameter is added but it does not exists.  For example:

```bicep
var bicep1 = builder.AddBicepTemplate("test", "test.bicep")
                   .WithParameter("paramFoo", parameter)
```

Can we make .netAspire to check if `paramFoo` exists on `test.bicep` ?
Can you just load the `test.bicep` as string to a variable and run `string.Contains(bicepString, "param paramFoo") and fail if that's false?

That would help users a lot to quickly detect typos; instead of waiting until azd is trying to compile the output bicep file.

Reply:
Too hacky. We could builld an analyzer but we still need a runtime check.

Reply:
There is also no guarantee that the file exists at compile time.

Reply:
This still needs a comment though. You can say it's from the bicep bnf